### PR TITLE
Add integration with vnc console

### LINKED ISSUE #490: Support web-based console connection
In addition to to .vv file we could support also web-based clients.

[(GS) edit: uses [patternfly-react VncConsole](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-3/react-console) ] 

Designs: https://docs.google.com/document/d/1m-pM0VVgeZmVCJFs2lLuzC9KUVV-X1cDfLXPboMWuBA/edit

selection_265

selection_266

### PR DESCRIPTION
**Pull request by kozloffsky**
Add integration with [VncConsole](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-3/react-console)

fixes #490

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by gregsheremeta** [comment]
ci add to whitelist

File: config/paths.js [hunk 1]
```diff
@@ -35,4 +35,5 @@ module.exports = {
   appNodeModules: resolveApp('node_modules'),
   ownNodeModules: resolveApp('node_modules'),
   nodePaths: nodePaths,
+  novmc: resolveApp('node_modules/@novnc/novnc')
```
Comment (gregsheremeta) [inline config/paths.js]: change to:

`novnc: resolveApp('node_modules/@novnc/novnc')`

('n' vs 'm')

Comment (bond95) [inline config/paths.js]: Is this module need? Cause I didn't see in https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-3/react-console `@novnc/novnc` requirement.

Comment (bond95) [reply config/paths.js]: Ok, I found it, but it already in module dependencies and question is still relevant. Is this module need to install explicitly?

Comment (kozloffsky) [reply config/paths.js]: @bond95 This module does not have in dist regular js code so it needs to me babelized.

File: src/App.js [hunk 2]
```diff
@@ -51,7 +51,9 @@ const App = ({ history, config, appReady }) => {
...
...MVFSVTB0UTBFd2daSUdBMVVkSXdTQmlqQ0JoNEFVVmllZnFjK2djSFVBQ2dDRThsVVFRQUxTTldHaGE2UnBcclxuTUdjeEN6QUpCZ05WQkFZVEFsVlRNU1F3SWdZRFZRUUtEQnR5YUdWMkxteGhZaTVsYm1jdVluSnhMbkpsWkdoaGRDNWpiMjB4TWpBd1xyXG5CZ05WQkFNTUtXSnljUzFrWlhZdWNtaGxkaTVzWVdJdVpXNW5MbUp5Y1M1eVpXUm9ZWFF1WTI5dExqUTNNVFk0Z2dJUUFEQUpCZ05WXHJcbkhSTUVBakFBTUE0R0ExVWREd0VCL3dRRUF3SUZvREFnQmdOVkhTVUJBZjhFRmpBVUJnZ3JCZ0VGQlFjREFRWUlLd1lCQlFVSEF3SXdcclxuTGdZRFZSMFJCQ2N3SllJalluSnhMV1JsZGk1eWFHVjJMbXhoWWk1bGJtY3VZbkp4TG5KbFpHaGhkQzVqYjIwd0RRWUpLb1pJaHZjTlxyXG5BUUVMQlFBRGdnRUJBRWI1V1NvU3BPd1JUZjdVNnlPLzdwNUIvT3RiZHQrdGtPei9WSFBhRm54YlcwTUMrR3hGdEI0dVBQNVNwVHNQXHJcblZkWDJ0aVRVeXlKU3RMbXRMVWxlTHYrM0ZVbW9NcDV2eis4UnU0REE4QzNzdlhrYkNTZS9Qd2JOOEFyYlN2eXI5aUMveUpmN25OQ2Rcclxuckh1bzEzWnpZNDEvdTlUSHhiSW5pNHZPT00xTGw0WDBwOVRSTko1bzBhSTl5djBoR2NTV21HT0QyUi9pNWk1SkJFMVFFaVVrVlJWUVxyXG5vS3l5QXVveExtNkQ4bkJKMWRrUzhHVUtuM3J4blFFUGNZd2I0dGdaS3V6bG03RkdMOWI3ZEw2ZWpyNStmTEdLY3ZYWkJXcmF1dXQwXHJcbkV0WDQ2ODI1elorZHg1NXpPbm9Xa2cvSzBDQkpjM0prQ29ES0toVlVZQTZTRGtZbGF1Zz1cclxuLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLVxuIiwic2lnbmVkRmllbGRzIjoic2FsdCxkYXRhLGRpZ2VzdCx2YWxpZEZyb20sdmFsaWRUbyIsInZhbGlkRnJvbSI6IjIwMTgxMTEyMTQyOTI4IiwidmFsaWRUbyI6IjIwMTgxMTEyMTQzMTI4In0'
```
Comment (gregsheremeta) [inline src/App.js]: what does `path` do and what is this giant string?

File: src/App.js [hunk 3]
```diff
@@ -51,7 +51,9 @@ const App = ({ history, config, appReady }) => {
...
+          host='brq-dev.rhev.lab.eng.brq.redhat.com' port='6100' />
```
Comment (gregsheremeta) [inline src/App.js]: these should be variables

File: src/components/VmActions/ConsoleConfirmationModal.js [hunk 4]
```diff
@@ -78,6 +78,6 @@ export default connect(
   (dispatch, { vm, consoleId }) => ({
     onCheckConsoleSessionInUse: ({ usbFilter, userId }) => dispatch(checkConsoleInUse({ vmId: vm.get('id'), usbFilter, userId })),
     onConsoleSessionConfirmaClose: () => dispatch(setConsoleInUse({ vmId: vm.get('id'), consoleInUse: null })),
-    onDownloadConsole: ({ usbFilter }) => dispatch(downloadConsole({ vmId: vm.get('id'), usbFilter, consoleId })),
+    onDownloadConsole: ({ usbFilter }) => dispatch(openConsole({ vmId: vm.get('id'), usbFilter: usbFilter, consoleId: consoleId })),
```
Comment (gregsheremeta) [inline src/components/VmActions/ConsoleConfirmationModal.js]: may want to rename `onDownloadConsole` to `onOpenConsole`

File: src/App.js (hunk 3 above)
Comment (bond95) [reply src/App.js]: +1

File: src/saga/console/index.js [hunk 5]
```diff
@@ -63,6 +63,33 @@ export function* downloadVmConsole (action) {
...
+      }
+    }
+  }
+  let data = yield callExternalAction('consoleProxyTicket', Api.consoleProxyTicket,
+    { type: 'INTRENAL_CONSOLE', payload: { vmId, consoleId } })
+  let ticket = yield callExternalAction('consoleTicket', Api.consoleTicket,
+    { type: 'INTRENAL_CONSOLE', payload: { vmId, consoleId } })
+  data.ticket = ticket
+  data.isSpice = isSpice
+  return data
```
Comment (bond95) [inline src/saga/console/index.js]: I don't really understand why do you return here `data`?

File: src/saga/console/index.js [hunk 6]
```diff
@@ -63,6 +63,33 @@ export function* downloadVmConsole (action) {
...
+    if (consolesInternal && consolesInternal.length > 0) {
+      let console = consolesInternal.find(c => c.protocol === 'spice') || consolesInternal[0]
+      consoleId = console.id
+      if (console.protocol === 'spice') {
+        isSpice = true
+      }
+    }
+  }
+  let data = yield callExternalAction('consoleProxyTicket', Api.consoleProxyTicket,
+    { type: 'INTRENAL_CONSOLE', payload: { vmId, consoleId } })
```
Comment (bond95) [inline src/saga/console/index.js]: Use constants, please.

**Comment by bond95** [comment]
@kozloffsky rebase, please

**Comment by gregsheremeta** [comment]
@kozloffsky i see you rebased, but you didn't squash. It is not reviewable until you do. The commits are a mess and impossible to follow.

*please* squash before you do any more work on this. If you don't know what I'm talking about, please ping me or @bond95 .

**Comment by gregsheremeta** [comment]
much better, thank you

**Comment by gregsheremeta** [comment]
ci build please

**Comment by gregsheremeta** [comment]
[needs ovirt-engine-nodejs-modules update for react-console]

**Comment by gregsheremeta** [comment]
@kozloffsky I couldn't get the latest commits to run.

Production compile and installed rpm:
selection_332

Dev compile:
selection_327
selection_328
selection_329
selection_330
selection_331

**Comment by gregsheremeta** [comment]
closing, doesn't work

**Pull request closed without merging**